src/fp.rs

```diff
@@ -257,6 +259,10 @@ impl FieldParameters {
         }
         assert_eq!(&self.roots, &roots[..MAX_ROOTS + 1], "roots mismatch");
         assert_eq!(self.from_elem(self.roots[0]), 1, "first root is not one");
+
+        let bit_mask: u128 = (1 << (129 - p.leading_zeros())) - 1;
```

Comment:
This will not work for a 127-bit or 128-bit prime.

Reply:
That's true, but the current implementation only allows up to 126. This is enforced by a check above:
```rust
if let Some(x) = p.checked_next_power_of_two() {
    if x > 1 << 126 {
        panic!("p >= 2^126");
    }
} else {
    panic!("p >= 2^126");                                                                                                                                        
}
```